miniapp/miniapp_bt_band_to_tridiag.cpp

```diff
@@ -0,0 +1,198 @@
+
+    Communicator world(MPI_COMM_WORLD);
+    CommunicatorGrid comm_grid(world, opts.grid_rows, opts.grid_cols, Ordering::ColumnMajor);
+
+    // Allocate memory for the matrix
+    GlobalElementSize matrix_size(opts.m, opts.n);
+    TileElementSize block_size(opts.mb, opts.nb);
+
+    ConstHostMatrixType matrix_ref = [matrix_size, block_size, comm_grid]() {
+      using dlaf::matrix::util::set_random_hermitian;
+
+      HostMatrixType hermitian(matrix_size, block_size, comm_grid);
+      set_random_hermitian(hermitian);
+
+      return hermitian;
+    }();
+
+    for (int64_t run_index = -opts.nwarmups; run_index < opts.nruns; ++run_index) {
+      if (0 == world.rank() && run_index >= 0)
+        std::cout << "[" << run_index << "]" << std::endl;
+
+      HostMatrixType mat_e_host(matrix_size, block_size, comm_grid);
+      copy(matrix_ref, mat_e_host);
+
+      HostMatrixType mat_hh_host({opts.m, opts.m}, {opts.mb, opts.mb}, comm_grid);
+
+      double elapsed_time;
+      {
+        MatrixMirrorType mat_e(mat_e_host);
+        MatrixMirrorType mat_hh(mat_hh_host);
+
+        // Wait for matrices to be copied to GPU (if necessary)
+        mat_e.get().waitLocalTiles();
+        mat_hh.get().waitLocalTiles();
+        DLAF_MPI_CHECK_ERROR(MPI_Barrier(world));
+
+        dlaf::common::Timer<> timeit;
+        dlaf::eigensolver::backTransformationBandToTridiag<Backend::MC>(opts.b, mat_e.get(),
+                                                                        mat_hh.get());
+
+        // wait and barrier for all ranks
+        mat_e.get().waitLocalTiles();
+        DLAF_MPI_CHECK_ERROR(MPI_Barrier(world));
+
+        elapsed_time = timeit.elapsed();
+      }
+
+      double gigaflops;
+      {
+        const double m = mat_e_host.size().rows();
+        const double n = mat_e_host.size().cols();
+        auto add_mul = m * m * n;
+        gigaflops = dlaf::total_ops<T>(add_mul, add_mul) / elapsed_time / 1e9;
+      }
+
+      // print benchmark results
+      if (0 == world.rank() && run_index >= 0)
+        std::cout << "[" << run_index << "]"
+                  << " " << elapsed_time << "s"
+                  << " " << gigaflops << "GFlop/s"
+                  << " " << dlaf::internal::FormatShort{opts.type} << " " << mat_e_host.size() << " "
+                  << mat_e_host.blockSize() << " " << opts.b << " " << comm_grid.size() << " "
+                  << pika::get_os_thread_count() << " " << backend << std::endl;
+
+      // (optional) run test
+      if ((opts.do_check == dlaf::miniapp::CheckIterFreq::Last && run_index == (opts.nruns - 1)) ||
+          opts.do_check == dlaf::miniapp::CheckIterFreq::All) {
+        DLAF_UNIMPLEMENTED("Check");
+      }
+    }
+  }
+};
+
+int pika_main(pika::program_options::variables_map& vm) {
+  {
+    dlaf::ScopedInitializer init(vm);
+    const Options opts(vm);
+
+    dlaf::miniapp::dispatchMiniapp<BacktransformBandToTridiagMiniapp>(opts);
+  }
+
+  return pika::finalize();
```

Comment:
Not a must for this PR as this can be updated elsewhere as well, but:
```suggestion
  pika::scoped_finalize fin;
  dlaf::ScopedInitializer init(vm);
  const Options opts(vm);

  dlaf::miniapp::dispatchMiniapp<BacktransformBandToTridiagMiniapp>(opts);

  return EXIT_SUCCESS;
```

Reply:
Is this a new pika feature?

Reply:
> Is this a new pika feature?

Yes, and I realize this was never mentioned in the release notes so no wonder you wouldn't know about it... It was added in 0.2.0.

Reply:
Small proposal: since this applies to almost all miniapps, I would open a separate PR with all required changes.

Moreover, in this future PR, I would propose also to have something like this

https://github.com/eth-cscs/DLA-Future/blob/89ca01fe3f7c365b34ce1d6d2adb78d79fde7c0e/miniapp/miniapp_bt_band_to_tridiag.cpp#L178-L179

where the name of the mini app is not hard-coded. It may be an overkill, in case I would just check that all names of miniapps are correct (I've seen at least one wrong name).

Reply:
Proposal accepted :wink: